tensorflow/core/kernels/segment_reduction_ops_impl.h

```diff
@@ -957,6 +957,125 @@ class SparseSegmentSqrtNGradOp
+                      done);
+    const int64 num_indices = indices.NumElements();
+
+    OP_REQUIRES_ASYNC(
+        context, TensorShapeUtils::IsVector(segment_ids.shape()),
+        errors::InvalidArgument("segment_ids should be a vector."), done);
+
+    OP_REQUIRES_ASYNC(
+        context, num_indices == segment_ids.NumElements(),
+        errors::InvalidArgument("indices and segment_ids should have"
+                                "same length."),
+        done);
+
+    ScratchSpace<Index> output_rows_host(context, 1, /* on_host */ true);
+    auto stream = context->op_device_context()->stream();
+
+    if (has_num_segments) {
+      const Tensor& num_segments = context->input(3);
+      se::DeviceMemoryBase num_segments_device(
+          const_cast<Tensor&>(num_segments).template flat<Index>().data()
+      );
+      OP_REQUIRES_ASYNC(
+          context,
+          stream
+              ->ThenMemcpy(output_rows_host.mutable_data(), num_segments_device, sizeof(Index)).ok(),
+          errors::Internal(
+              "SparseSegmentSumGpuOp: failed to copy num_segments to host."),
+          done);
+    } else {
+      se::DeviceMemoryBase last_segment_id_on_device(
+          const_cast<Tensor&>(segment_ids).template flat<Index>().data() +
+          (num_indices - 1));
+      OP_REQUIRES_ASYNC(
+          context,
+          stream
+              ->ThenMemcpy(output_rows_host.mutable_data(), last_segment_id_on_device,
+                           sizeof(Index))
+              .ok(),
+          errors::Internal(
+              "SparseSegmentSumGpuOp: failed to copy output_rows to host."),
+          done);
+    }
+
+    const Index input_dims = input_data.dims();
+    OP_REQUIRES_ASYNC(
+        context, input_dims >= 1,
+        errors::InvalidArgument("indices and segment_ids should have "
+                                "same length."),
+        done);
+
+#define ACCUMULATE_MUL(SHAPE, RANGE, ACCUM) \
+  for (Index _i = 1; _i < RANGE; _i++) {    \
+    ACCUM *= SHAPE.dim_size(_i);            \
+  }                                         \
+  while (0)
+    Index accum = 1;
+    const TensorShape input_shape = input_data.shape();
+    Index element_size_muta = 1;
+    if (input_dims > 1) {
+      ACCUMULATE_MUL(input_shape, input_dims, accum);
+      element_size_muta = accum;
+    }
+    const Index element_size = element_size_muta;
+
+    functor::SparseSegmentSumFunctor<T, Index> functor_;
+
+    auto create_and_check_output = [context, num_indices, element_size,
+                                    output_rows_host, &input_data, &indices,
+                                    &segment_ids, &functor_, done]() {
+      auto stream = context->op_device_context()->stream();
+      ScopedActivateExecutorContext scoped_activation{stream->parent()};
+
+      Index output_rows = *output_rows_host.data();
+      if (! has_num_segments) {
+        output_rows++;
+      }
```

Comment:
Please add a short comment on why this is needed.

Reply:
Thanks for your advice. I've added the [comment](https://github.com/tensorflow/tensorflow/pull/45085/commits/13c00d7463afd62c6a5425ad35aaea1e4c377d63#diff-d29812caca45335066ddacf8fb06afc416f1f557bc446068ebaa163495e0b03bR1024).